### PR TITLE
GH-1591 Add support for PEM

### PR DESCRIPTION
**Pull request by zrdzn**

### CODE REVIEW

**Comment by zugazagoitia** [comment]
Stable ssl-plugin is out, tested against Javalin 5.1.0

**Comment by zrdzn** [comment]
Review where 🐒

File: reposilite-backend/src/main/kotlin/com/reposilite/web/application/JavalinConfiguration.kt [hunk 1]
```diff
@@ -146,20 +147,28 @@ internal object JavalinConfiguration {
...
+                it.securePort = localConfiguration.sslPort.get()
 
-            if (!localConfiguration.enforceSsl.get()) {
-                val standardConnector = ServerConnector(server)
-                standardConnector.port = localConfiguration.port.get()
-                server.addConnector(standardConnector)
+                val keyPath = localConfiguration.keyPath.get().replace("\${WORKING_DIRECTORY}", reposilite.parameters.workingDirectory.toAbsolutePath().toString())
+                val keyPassword = localConfiguration.keyPassword.get()
+
+                if (keyPath.endsWith(".pem")) {
```
Comment (dzikoysk) [inline reposilite-backend/src/main/kotlin/com/reposilite/web/application/JavalinConfiguration.kt:160]: Replace `if` statement with `when` and throw on `else`.

File: reposilite-backend/src/main/kotlin/com/reposilite/web/application/JavalinConfiguration.kt [hunk 2]
```diff
@@ -146,20 +147,28 @@ internal object JavalinConfiguration {
...
+                    val paths = keyPath.split(" ")
```
Comment (dzikoysk) [inline reposilite-backend/src/main/kotlin/com/reposilite/web/application/JavalinConfiguration.kt:161]: val (certPath, keyPath) = path.split

File: reposilite-backend/src/main/kotlin/com/reposilite/configuration/local/LocalConfiguration.kt [hunk 3]
```diff
@@ -84,12 +84,14 @@ class LocalConfiguration : Facade {
     @Description("# SSL port to bind")
     val sslPort = reference(443)
 
-    @Description("# Key store file to use.")
+    @Description("# Key file to use.")
     @Description("# You can specify absolute path to the given file or use \${WORKING_DIRECTORY} variable.")
-    val keyStorePath = reference("\${WORKING_DIRECTORY}/keystore.jks")
+    @Description("# If you want to use .pem certificate you need to specify its path next to the key path:")
+    @Description("# keyPath: \${WORKING_DIRECTORY}/key.pem \${WORKING_DIRECTORY}/cert.pem")
```
Comment (dzikoysk) [inline reposilite-backend/src/main/kotlin/com/reposilite/configuration/local/LocalConfiguration.kt:90]: Add example with JKS file

**Review by dzikoysk — CHANGES_REQUESTED** [review]

**Review by dzikoysk — APPROVED** [review]

**Comment by dzikoysk** [comment]
Thanks!

**Pull request merged**